### PR TITLE
Proposals didc governance gen

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
We would like to free ourselves of the dependency tree of other canisters, such as the `nns_governance` canister.

# Changes
- Add code to derive Rust types exposed by the governance canister.

# Tests
- None yet.  This code does not even include the generated code as a module.

# Todos

- [x] Add entry to changelog (if necessary).
  - I suggest adding a changelog entry when we switch to using this generated code.

### CODE REVIEW

File: rs/proposals/src/canisters/nns_governance/api.rs
Comment (bitdivine) [inline rs/proposals/src/canisters/nns_governance/api.rs:1]: This is the main file of this PR - the generated Rust code.

File: scripts/proposals/did2rs
Comment (bitdivine) [inline scripts/proposals/did2rs:1]: This is the command that generates the Rust code, and that would be used to regenerate the Rust code if it is updated.

File: scripts/proposals/did2rs-list
Comment (bitdivine) [inline scripts/proposals/did2rs-list:1]: Currently we generate only `nns_governance` but more canisters are coming.  This gets a list of the canisters we can generate.  I optimistically assume that we will import everything we can.  If that turns out to be over-optimistic, we can print a hard-wired list of canisters to generate.

**Review by mstrasinskis — APPROVED** [review]
Rubber stamp.

**Pull request merged**